Observer/Checkout/RedirectToBoldCheckoutObserver.php

```diff
@@ -70,7 +77,9 @@ public function __construct(
         LoggerInterface $logger,
         ClientInterface $client,
         IsOrderInitializationAllowedInterface $isOrderInitializationAllowed,
-        IsRedirectToBoldCheckoutAllowedInterface $isRedirectToBoldCheckoutAllowed
+        IsRedirectToBoldCheckoutAllowedInterface $isRedirectToBoldCheckoutAllowed,
+        private readonly BoldQuoteRepositoryInterface $boldQuoteRepository,
+        private readonly ResumeOrder $resumeOrder
```

Comment:
Backwards compatibility: Constructor property promotion and the `readonly` modifier are PHP 8 features that will break compatibility with PHP 7 (we support back to 7.2). 

**Hint**: There is a [Rector](https://getrector.com) configuration file in the root that will check for PHP compatibility issues. Note that syntax formatting will need to be adjusted to match PSR-12 or PER-CS after running this tool.